### PR TITLE
Feature/week1

### PR DESCRIPTION
**Pull request by Garett601**
# Description

Create a Python package that defines preprocessing steps for your chosen dataset so that it can be used in a machine learning project.

create a feature branch in your personal repository, and create a PR to main for us to review your code.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Test utils code
- [x] Test data preprocessing code
- [x] Test model code

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Release Notes

- **New Features**
  - Introduced a new pull request template for standardized contributor submissions.
  - Added configuration files for project settings and Databricks integration.
  - Enhanced power consumption data processing with new classes and methods for model training and evaluation.
  - Improved ownership clarity in the repository with updated CODEOWNERS file.
  - Added new utility functions for visualizing power consumption data.

- **Documentation**
  - Updated README with clearer setup instructions and additional troubleshooting tips.
  - Improved documentation for configuration management and data schemas.

- **Bug Fixes**
  - Addressed issues with environment setup and Python version compatibility.

- **Tests**
  - Added comprehensive unit tests for the `ConsumptionModel` and `DataProcessor` classes, ensuring robust functionality.
  - Established testing frameworks for utility functions related to data visualization.

- **Chores**
  - Updated project metadata and dependencies in `pyproject.toml`.
  - Enhanced `.gitignore` to exclude additional files and directories.

### CODE REVIEW

File: databricks.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+bundle:
+  name: mlops-with-databricks-bundle
+
+include:
+  - resources/*.yml
+
+artifacts:
```
Comment (Garett601) [inline databricks.yml:7]: I will need to change this once we get to that module

**Comment by basakeskili** [comment]
Looks great!

Small question: were you able to package your module using uv lock?

**Comment by Garett601** [comment]
> Looks great!
> 
> Small question: were you able to package your module using uv lock?

I am going to fight that battle tomorrow 😆 I am not using src structure, I have added a few things to my pyproject.toml to make sure that it packages the correct directory... But I will investigate if it worked and if I need to change things tomorrow.

Main reason for not using src structure is that our current projects don't use source, and I need to figure this out if I want to apply it to our projects.... or I just switch over to poetry that handles this well.

**Comment by Garett601** [comment]
> Looks great!
> 
> Small question: were you able to package your module using uv lock?

seems to work as expected with `uv lock` and `uv build` 
image

File: databricks.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+bundle:
+  name: mlops-with-databricks-bundle
+
+include:
+  - resources/*.yml
+
```
[the remark this answers is not in the document]
Comment (Garett601) [reply databricks.yml:6]: Resources will be added in the coming PR's

File: power_consumption/model/rf_model.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
+"""
+Main module for the power consumption model.
+"""
+
+from typing import Dict, Self, Tuple
```
[the remark this answers is not in the document]
Comment (Garett601) [reply power_consumption/model/rf_model.py:5]: Project needs to be ~3.11 for DBR 15.4 LTS runtime deployments

**Review by basakeskili — APPROVED** [review]

**Pull request merged**